submissions/andrewklmn/friends-app/app.js

```diff
@@ -0,0 +1,358 @@
+  filters.ageRange[1] = filterMaxAge.value = maxPersonAge;
+  filters.country = filterCountry.value = '';
+
+  sorter.keyName = sortField.value = '';
+  sortField.dispatchEvent(change);
+}
+
+const redrawFriends = (state) => {
+  const {filters, sorter, initialListLength} = state;
+  preloader.classList.remove('hidden');
+  setResetButtonView(filters, sorter);
+  state.numberOfShowedFriends = 0;
+  container.innerHTML = '';
+  container.scrollTop = 0;
+
+  sortList(filterList(state))
+    .slice(0, initialListLength)
+    .forEach((friend) => {
+      state.numberOfShowedFriends++;
+      drawPerson(friend);
+    });
+
+  preloader.classList.add('hidden');
+}
+
+const updateFriendsList = (list)=> {
+  state.friends = list;
+  redrawFriends(state);
+}
+
+const addOptionToSelect = (select, optionValue, optionText) => {
+  const option = document.createElement('option');
+  option.value = optionValue;
+  option.innerHTML = optionText;
+  select.appendChild(option);
+}
+
+const drawSorter = (state)=>{
+  const {sorter} = state;
+  Object.keys(sorter.fieldPathList).forEach((field)=>{
+    addOptionToSelect(sortField, field, field);
+  });
+}
+
+const drawFilters = (state) => {
+
+  const onlyUnique = (value, index, self) => {
+    return self.indexOf(value) === index;
+  };
+  const {friends, filters} = state;
+
+  filters.genderList = friends.map(person => person.gender).filter(onlyUnique);
+  filters.genderList.forEach((gender)=>{
+    addOptionToSelect(filterGender, gender, gender);
+  });
+  filterMinAge.value = filters.ageRange[0];
+  filterMaxAge.value = filters.ageRange[1];
+  filters.countryList = friends.map(person => person.location.country).filter(onlyUnique);
+  filters.countryList.sort().forEach( country => {
+    addOptionToSelect(filterCountry, country, country);
+  });
+};
+
+const initApp = (state) => {
+
+  fetch(randomUserUrl)
+    .then(response => response.json())
+    .then(({results}) => {
+      updateFriendsList(results);
+      drawSorter(state);
+      drawFilters(state);
+      preloader.classList.add('hidden');
+    })
+    .catch(function() {
+      const header = document.querySelector('.header');
+      const footer = document.querySelector('.footer')
+      header.innerHTML = "ERROR: Can't get friend list!";
+      header.classList.add('error');
+      footer.innerHTML = "Try Refresh this page later!";
+      footer.classList.add('error');
+    });
+};
+
+const drawMoreFriends = (friends) => {
+  
+  friends.forEach((friend) => {
+    state.numberOfShowedFriends++;
+    drawPerson(friend);
+  });
+  
+  setTimeout(() => {
+    if (container.scrollTop > 0) {
+      container.scrollTop += state.afterLoadingMoreFriendsScrollShift;
+    };
+  }, state.nextMoreFriendAutoloadDelay * 2);
+};
+
+const autoLoaderOnScroll = (state) => {
+  const {numberOfShowedFriends, friends, initialListLength, nextMoreFriendAutoloadDelay} = state;
+  
+  if (state.scrollDisabled) return true;
```

Comment:
Do you really need return something in this function? If you just want to use early return technique, you can use `if (state.scrollDisabled) return`. It returns `false` by default.

Reply:
You are right! 
I used it automatically because in JQuery return false; means a.stopPropagation()...
But in this case, when I use addEventListener, I don't need it. Because 'return false;' doesn't stop event